src/Npgsql/Internal/NpgsqlWriteBuffer.cs

```diff
@@ -581,5 +580,13 @@ internal byte[] GetContents()
         return buf;
     }
 
+    [Conditional("DEBUG")]
+    internal static void AssertASCIIOnly(string s)
+    {
+        foreach (var c in s)
```

Comment:
I'm guessing LINQ usage doesn't matter in `[Conditional("DEBUG")]` (not that this code really needs it though)

Reply:
Correct it just made it easier to make sure we don't even open System.Linq anywhere.

Reply:
If only we had an analyzer for that...